src/core/tests/pass/serialization/deterministicity.cpp

```diff
@@ -130,3 +145,150 @@ TEST_F(SerializationDeterministicityTest, ModelWithConstants) {
+    std::ifstream xml_2(m_out_xml_path_2, std::ios::in | std::ios::binary);
+    EXPECT_TRUE(files_equal(xml_1, xml_2));
+}
+
+TEST_P(SerializationDeterministicityInputOutputTest, FromIrModel) {
+    auto irVersion = GetParam();
+
+    std::string irModel_1stPart = R"V0G0N(<?xml version="1.0"?>
+<net name="Model0" version=")V0G0N";
+    std::string irModel_2ndPart = R"V0G0N(">
+	<layers>
+		<layer id="0" name="input0" type="Parameter" version="opset1">
+			<data shape="1,3,22,22" element_type="f32" />
+			<output>
+				<port id="0" precision="FP32">
+					<dim>1</dim>
+					<dim>3</dim>
+					<dim>22</dim>
+					<dim>22</dim>
+				</port>
+			</output>
+		</layer>
+		<layer id="1" name="input1" type="Parameter" version="opset1">
+			<data shape="1,3,22,22" element_type="f32" />
+			<output>
+				<port id="0" precision="FP32">
+					<dim>1</dim>
+					<dim>3</dim>
+					<dim>22</dim>
+					<dim>22</dim>
+				</port>
+			</output>
+		</layer>
+		<layer id="2" name="output0" type="Result" version="opset1">
+			<input>
+				<port id="0" precision="FP32">
+					<dim>1</dim>
+					<dim>3</dim>
+					<dim>22</dim>
+					<dim>22</dim>
+				</port>
+			</input>
+		</layer>
+		<layer id="3" name="output1" type="Result" version="opset1">
+			<input>
+				<port id="0" precision="FP32">
+					<dim>1</dim>
+					<dim>3</dim>
+					<dim>22</dim>
+					<dim>22</dim>
+				</port>
+			</input>
+		</layer>
+	</layers>
+	<edges>
+		<edge from-layer="0" from-port="0" to-layer="2" to-port="0" />
+		<edge from-layer="1" from-port="0" to-layer="3" to-port="0" />
+	</edges>
+	<rt_info />
+</net>
+)V0G0N";
+    std::string strVersion = irVersion == ov::pass::Serialize::Version::IR_V11 ? "11" : "10";
+    std::string irModel = irModel_1stPart + strVersion + irModel_2ndPart;
+
+    {
+        std::ofstream xmlFile;
+        xmlFile.open(xmlFileName);
+        xmlFile << irModel;
+        xmlFile.close();
+    }
+
+    auto expected1 = ov::test::readModel(xmlFileName, "");
+    ov::pass::Serialize(m_out_xml_path_1, "", irVersion).run_on_model(expected1);
+    auto expected2 = ov::test::readModel(m_out_xml_path_1, "");
+    ov::pass::Serialize(m_out_xml_path_2, "", irVersion).run_on_model(expected2);
+
+    EXPECT_EQ(input0Name, expected1->input(0).get_node()->get_friendly_name());
+    EXPECT_EQ(input1Name, expected1->input(1).get_node()->get_friendly_name());
+    EXPECT_EQ(output0Name, expected1->output(0).get_node()->get_friendly_name());
+    EXPECT_EQ(output1Name, expected1->output(1).get_node()->get_friendly_name());
+    EXPECT_EQ(input0Name, expected2->input(0).get_node()->get_friendly_name());
+    EXPECT_EQ(input1Name, expected2->input(1).get_node()->get_friendly_name());
+    EXPECT_EQ(output0Name, expected2->output(0).get_node()->get_friendly_name());
+    EXPECT_EQ(output1Name, expected2->output(1).get_node()->get_friendly_name());
+
+    std::ifstream initial_xml(xmlFileName, std::ios::in | std::ios::binary);
```

Comment:
The part below is look like a test on `Serialization`. Is it not covered by serialization tests?